src/components/Courses/Courses.jsx

```diff
@@ -1,38 +1,50 @@
-import React, { useEffect, useState } from 'react';
+import React, { useEffect } from 'react';
 import { useNavigate } from 'react-router-dom';
+import { useSelector } from 'react-redux';
 
 import CourseCard from './components/CourseCard/CourseCard';
 import SearchBar from './components/SearchBar/SearchBar';
 import Button from '../../common/Button/Button';
 import dataFormat from '../../helpers/dataFormat';
-import { getCourses } from '../../services/courseService';
+import { getAuthors, getCourses, searchCourses } from '../../services';
 import { ADD_COURSES, ROUTE_COURSES_ADD, ROUTE_LOGIN } from '../../constants';
-import isLoggedIn from '../../helpers/checkLogIn';
+import getCoursesSelector from '../../store/courses/selectors';
+import getAuthorsSelector from '../../store/authors/selectors';
+import getUsersSelector from '../../store/user/selectors';
+import getAuthorName from '../../helpers/getAuthorName';
 import styles from './styles.module.scss';
 
 const Courses = () => {
-	const [courses, setCourses] = useState([]);
+	const userData = useSelector(getUsersSelector);
+	const coursesData = useSelector(getCoursesSelector);
+	const authorsData = useSelector(getAuthorsSelector);
 
+	const allCourses = coursesData?.list;
+	const searchResult = coursesData?.searchResult;
+	const availableAuthors = authorsData?.list;
 	let navigate = useNavigate();
 
 	useEffect(() => {
-		if (!isLoggedIn()) {
+		if (!userData.isAuth) {
```

Comment:
If i refresh the page, i will be navigated to login page even though i have a token stored in localStorage